numpy/linalg/tests/test_linalg.py

```diff
@@ -1583,8 +1583,45 @@ def check_qr(self, a):
         assert_almost_equal(r2, r1)
 
     def test_qr_empty(self):
-        a = np.zeros((0, 2))
-        assert_raises(linalg.LinAlgError, linalg.qr, a)
+        for m, n in [(3, 0), (0, 3)]:
+            k = min(m, n)
+            a = np.empty((m, n))
+            a_type = type(a)
+            a_dtype = a.dtype
+            #
+            q, r = np.linalg.qr(a, mode='reduced')
+            assert_(q.dtype == a_dtype)
+            assert_(r.dtype == a_dtype)
```

Comment:
Better to use `assert_equal` for these cases, which produces a better error message, I think